[5G: MVC][component: agw][type: enhancement] AMF-Core

chandra-77 opened the pull request:
Signed-off-by: chandra-77 <dangudubiyyam.c@altencalsoftlabs.com>

<!--
    Tag your PR title with the components that it touches.
    E.g. "[lte][agw] Changeset" or "[orc8r][docker] ..."
-->

## Summary

<!-- Enumerate changes you made and why you made them -->
AMF core base code is added.
## Test Plan

<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->
Integ_tests are not done, because this PR is depend on NGAP, NAS5G, SCTP PRs.. 
## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

chandra-77 committed:
AMF-Core

Signed-off-by: chandra-77 <dangudubiyyam.c@altencalsoftlabs.com>

sureshgorijavolu commented on lte/gateway/c/oai/tasks/amf/CMakeLists.txt:
```diff
@@ -0,0 +1,81 @@
+
+target_compile_definitions(TASK_AMF_APP PRIVATE
+  PACKAGE_NAME=\"AMF\"
+  PACKAGE_VERSION=\"0.1\"
```
is package version required?

sureshgorijavolu commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,878 @@
+
+using namespace std;
+#define M5GS_REGISTRATION_RESULT_MAXIMUM_LENGTH                                \
+  1  // TODO temporary should be in nas5g
```
per comment, are you looking to move out the macro to nas5g folder?

sureshgorijavolu commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,878 @@
+nas_proc nas_procedure_reg;
+amf_as_data_t amf_data_sec;
+m5g_authentication m5g_auth;
+identification amf_proc;
```
it is good to follow a uniform naming convention for the types

sureshgorijavolu commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,878 @@
+  int rc = RETURNerror;
+  OAILOG_INFO(
+      LOG_NAS_AMF,
+      "AMF_TEST Authentication procedure success and start  Security mode"
```
erase out AMF_TEST. Logging statements need to clearly communicate the intent.

sureshgorijavolu commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,878 @@
+      "AMF_TEST Authentication procedure success and start  Security mode"
+      "command procedures\n");
+  nas_amf_registration_proc_t* registration_proc =
+      nas_procedure_reg.get_nas_specific_procedure_registration(amf_context);
```
function name can be little more meaningful

sureshgorijavolu commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,878 @@
+}
+
+//------------------------------------------------------------------------------
+static int amf_start_registration_proc_authentication(
```
function name is longer. can this be pruned to a shorter and meaningful name?

sureshgorijavolu commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,878 @@
+    nas_amf_registration_proc_t* registration_proc) {
+  OAILOG_FUNC_IN(LOG_NAS_AMF);
+  int rc = RETURNerror;
+  OAILOG_INFO(LOG_AMF_APP, "AMF-TEST: , from %s\n", __FUNCTION__);
```
is this info log required?

sureshgorijavolu commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,878 @@
+  OAILOG_INFO(LOG_AMF_APP, "AMF-TEST: , from %s\n", __FUNCTION__);
+  if ((amf_context) && (registration_proc)) {
+    OAILOG_INFO(
+        LOG_AMF_APP, "AMF-TEST: , calling amf_proc_authentication() from %s\n",
```
take out AMF-TEST. you want this as DEBUG level log than as INFO level log?

sureshgorijavolu commented on lte/gateway/c/oai/tasks/amf/CMakeLists.txt:
```diff
@@ -0,0 +1,81 @@
+include_directories("${NAS5G_C_DIR}/include/ies")
+include_directories("${NAS_C_DIR}/include/nas")
+include_directories(${LTE_OUT_DIR})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../../../../../../orc8r/gateway/c/common/logging)
```
is this needed?

ulaskozat replied:
@chandra-77 do you need all the statements in this cmake file? Can you please confirm?

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_main.cpp:
```diff
@@ -0,0 +1,134 @@
+              .pdu_address.redirect_server_address);
+      for (int i = 0; (N11_CREATE_PDU_SESSION_RESPONSE(received_message_p)
+                           .pdu_address.redirect_server_address != '\0') &&
+                      (i < 6);
```
What is the meaning of 6 here? Provide a meaningful definition and explanation for such constants.

chandra-77 replied:
This for loop is only for our debuggig purpose to print the IP. Removed the loop and updated the File

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+      (smf_context->dl_session_ambr == 0) ||
+      (smf_context->dl_session_ambr == 0)) {
+    /* AMBR has not been populated till now and default assigned
+     * TODO rechaeck the default values 64kbps & 32mbps
```
typo `rechaeck`

chandra-77 replied:
taken care

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+}
+
+/*
+ * As per Baicel capture and message flow, the function to be called
```
remove reference to Baicells.

chandra-77 replied:
removed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+int pdu_session_resource_setup_request(
+    ue_m5gmm_context_s* ue_context, amf_ue_ngap_id_t amf_ue_ngap_id) {
+  if (!ue_context) {
+    // TODO
```
you cannot leave this as is. log an error message and return error code. If ue_context is nullptr, this will lead to segfault on line 108.

chandra-77 replied:
taken care

ulaskozat replied:
what is to be done further here since you still have the `// TODO` comment, I expect more explanation about what needs to be done and incomplete.

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+  ngap_pdu_ses_setup_req->ue_aggregate_maximum_bit_rate.dl = dl_pdu_ambr;
+  ngap_pdu_ses_setup_req->ue_aggregate_maximum_bit_rate.ul = ul_pdu_ambr;
+
+  // Hardcored number of pdu sessions as 1
```
Hardcored -> Hardcoded

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+          smf_context->smf_proc_data.pdu_session_identity.pdu_session_id;
+
+  /* preparing for  bstring PDU_Session_Resource_Setup_Transfer
+   * amf_pdu_ses_setup_transfer_req is the structure to be filed
```
filed -> filled

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+
+  /* preparing for  bstring PDU_Session_Resource_Setup_Transfer
+   * amf_pdu_ses_setup_transfer_req is the structure to be filed
+   * and make a bundle and converted to bstring
```
reword this comment.

chandra-77 replied:
taken care

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+  amf_pdu_ses_setup_transfer_req.pdu_aggregate_max_bit_rate.ul = ul_pdu_ambr;
+
+  // UPF tied 4 octet and respective ip address are from
+  // SMF date that written to context
```
change the comment to "SMF context"

chandra-77 replied:
changed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+  // SMF date that written to context
+  memcpy(
+      &amf_pdu_ses_setup_transfer_req.up_transport_layer_info.gtp_tnl.gtp_tied,
+      smf_context->gtp_tunnel_id.upf_gtp_teid, 4);
```
check with @sanjay-ACL, this field will be coming as uint32_t from SMF. Make sure that it was converted to network byte order.

chandra-77 replied:
we will update and raise a separate PR.

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+      smf_context->gtp_tunnel_id.upf_gtp_teid, 4);
+  amf_pdu_ses_setup_transfer_req.up_transport_layer_info.gtp_tnl
+      .endpoint_ip_address =
+      blk2bstr(&smf_context->gtp_tunnel_id.upf_gtp_teid_ip_addr, 4);
```
again check with @sanjay-ACL, as IP address will be coming as string from SMF side.

chandra-77 replied:
we will update and raise separate PR

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_level_qos_param.alloc_reten_priority.pre_emption_cap =
+
+      // MAY_TRIGGER_PRE_EMPTION;
```
move this comment to before assignment.

chandra-77 replied:
moved

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_level_qos_param.alloc_reten_priority.pre_emption_vul =
+
+      // PRE_EMPTABLE;
```
move this comment to before assignment. why is there an empty line?

chandra-77 replied:
taken care

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+          .qos_flow_level_qos_param.alloc_reten_priority.priority_level;
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_level_qos_param.alloc_reten_priority.pre_emption_cap =
+
```
why is there an empty line?

chandra-77 replied:
taken care

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+      "Converting pdu_session_resource_setup_request_transfer_t to bstring \n");
+  ngap_pdu_ses_setup_req->pduSessionResource_setup_list.item[0]
+      .PDU_Session_Resource_Setup_Request_Transfer =
+      amf_pdu_ses_setup_transfer_req;
```
I don't see any bstring here??

chandra-77 replied:
taken care

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+ * No NAS message is involved and direct itti_message is sent to NGAP.
+ * The transfer message is bstring and heavily nested structure is
+ * converted to bstring before sending to NGAP
+ */
```
I do not see any conversion to bstring in this function.

chandra-77 replied:
taken care

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+  return RETURNok;
+}
+
+/* Resourece release request to gNB through NGAP */
```
Typo 'Resourece'

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+int pdu_session_resource_release_request(
+    ue_m5gmm_context_s* ue_context, amf_ue_ngap_id_t amf_ue_ngap_id) {
+  if (!ue_context) {
+    // TODO
```
again don't leave this dangling. if it is nullptr, log error message and return error code.

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,261 @@
+      NORMAL_RELEASE;
+  amf_pdu_ses_rel_transfer_req.cause.cause_group.cause_group_type = NAS_GROUP;
+
+  // Convert amf_pdu_ses_setup_transfer_req to bstring and assign to message_p
```
what bstring are you referring to here?

chandra-77 replied:
resolved

ulaskozat commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,856 @@
+m5g_authentication m5g_auth;
+identification amf_proc;
+nas_network nas_nw;
+nas_amf_smc_proc_t smc_proc;
```
what is the reason why all these are dangling as global variables? if you are only using the methods from these classes without any object data, simply define these functions or use class scope, e.g., `amf_sap_c::send(...)`.

chandra-77 replied:
resolved

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_sap.h:
```diff
@@ -0,0 +1,214 @@
+
+using namespace std;
+#define MIN_GUMMEI 1
+#define MAX_GUMMEI 5
```
please comment on why these are the min and max values. what is the purpose.

chandra-77 replied:
resolved

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_sap.h:
```diff
@@ -0,0 +1,214 @@
+  ~amf_sap_t() {}
+  amf_primitive_t primitive;
+  primitive_t u;
+};
```
what is the purpose of each of these fields? I do not see any methods, looks like an overkill. Why not simply define this as a struct?

chandra-77 replied:
changed to struct

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_sap.h:
```diff
@@ -0,0 +1,214 @@
+  ~amf_sap_c() {}
+  void amf_sap_initialize(void);
+  int amf_sap_send(amf_sap_t* msg);
+};
```
this also looks like an over kill. I do not see any class variables, what are you going to initialize? I only see amf_sap_send method being used. Simply define this as a function; you do not need to define a class for one function unless you have plans to extend this class. If that is the case, please specify them as TO DO and provide a comment on extensions.

chandra-77 replied:
took the functions out and deleted the class.

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+ * ---------------------------------------------------------------------------
+ */
+class amf_context_t {
+ public:
```
why is this a class? simply define it as struct.

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+  bool is_imsi_only_detach;
+};
+
+class amf_ue_context_t {
```
why is this a class if you are not going to define proper constructors and destructors for the pointers.

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+  M5GCM_CONNECTED,
+};
+
+class ue_m5gmm_context_s {
```
again defining this as class does not make much sense if it has only public variables and no class methods.

chandra-77 replied:
changed to struct

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+ *  @brief Useful parameters to know in AMF application layer. They are set
+ * according to 3GPP TS.23.518 #6.1.6.2.25
+ */
+class ue_mm_context {
```
same class vs. struct comment.

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+  amf_ue_ngap_id_t amf_ue_ngap_id;
+};
+
+class amf_app_ue_context : public amf_ue_context_t, public ue_m5gmm_context_s {
```
inheritance is quite needless here. The inherited classes simple structs, define them as private class variables in `class amf_app_ue_context`.

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+      ue_m5gmm_context_s* const ue_context_p);
+};
+
+ue_m5gmm_context_s* amf_create_new_ue_context(void);
```
provide a comment for each and every function about its purpose.

chandra-77 replied:
resolved

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+
+ue_m5gmm_context_s* amf_create_new_ue_context(void);
+
+/* Retrive required UE context from respective hash table*/
```
leave single space before `*/`

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+
+/* Retrive required UE context from respective hash table*/
+amf_context_t* amf_context_get(const amf_ue_ngap_id_t ue_id);
+ue_m5gmm_context_s* amf_ue_context_exists_amf_ue_ngap_id(
```
provide a comment about the function.

chandra-77 replied:
resolved

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+
+void amf_ctx_set_attribute_present(
+    amf_context_t* ctxt, const int attribute_bit_pos) __attribute__((nonnull))
+__attribute__((flatten));
```
provide comments about the function usage

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+  uint8_t security_header_type;
+  uint8_t message_type;
+  uint8_t sequence_number;
+};
```
why is this a class?

chandra-77 replied:
changed to struct

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+ public:
+  void amf_app_ue_context_release(
+      ue_m5gmm_context_s* ue_context_p, ngap_Cause_t cause);
+};
```
why is this a class?

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+  uint8_t mac_matched : 1;
+  uint8_t security_context_available : 1;
+  int amf_cause;
+};
```
why is this a class?

chandra-77 replied:
resolved

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+  ~AMFMsg() {}
+
+  amf_msg_header header;
+  RegistrationRequestMsg registrationrequestmsg;
+  RegistrationAcceptMsg registrationacceptmsg;
+  RegistrationCompleteMsg registrationcompletemsg;
+  RegistrationRejectMsg registrationrejectmsg;
+  IdentityRequestMsg identityrequestmsg;
+  IdentityResponseMsg identityresponsemsg;
+  AuthenticationRequestMsg authenticationrequestmsg;
+  AuthenticationResponseMsg authenticationresponsemsg;
+  AuthenticationRejectMsg authenticationrejectmsg;
+  AuthenticationFailureMsg authenticationfailuremsg;
+  SecurityModeCommandMsg securitymodecommandmsg;
+  SecurityModeCompleteMsg securitymodecompletemsg;
+  SecurityModeRejectMsg securitymodereject;
+  DeRegistrationRequestUEInitMsg deregistrationequesmsg;
+  DeRegistrationAcceptUEInitMsg deregistrationacceptmsg;
+  ULNASTransportMsg uplinknas5gtransport;
+  DLNASTransportMsg downlinknas5gtransport;
```
unionize

chandra-77 replied:
unionized

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+typedef int (*sdu_out_not_delivered_t)(
+    amf_context_t* amf_ctx, nas_amf_proc_t* nas_proc);
+typedef int (*sdu_out_not_delivered_ho_t)(
+    amf_context_t* amf_ctx, nas_amf_proc_t* nas_proc);
```
please comment about the purpose of these typedefs

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+
+class nas5g_base_proc_t;
+class nas_amf_proc_t;
+class nas_amf_registration_proc_t;
```
why are these classes?

chandra-77 replied:
for forward declaration, changed to structs

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+
+  class nas5g_base_proc_t* parent;
+  class nas5g_base_proc_t* child;
+};
```
define as struct

chandra-77 replied:
defined as struct

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+  sdu_out_not_delivered_t not_delivered;
+  sdu_out_not_delivered_ho_t not_delivered_ho;
+  amf_fsm_state_t previous_amf_fsm_state;
+};
```
define as struct

chandra-77 replied:
defined as struct

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+ public:
+  nas_amf_proc_t amf_proc;
+  amf_specific_proc_type_t type;
+};
```
define as struct

chandra-77 replied:
defined as struct

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+
+class identification : public amf_context_t {
+ public:
+  char amf_identity_type_str[5][15] = {"NOT AVAILABLE", "IMSI", "IMEI",
```
why define this as 2D array rather than defining each constant string with `#define`

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+  amf_specific_proc_type_t type;
+};
+
+class identification : public amf_context_t {
```
this class definition, particularly the inheritance does not make much sense.

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+      imeisv_t* const imeisv, uint32_t* const tmsi);
+};
+
+class nas_amf_auth_proc_t {
```
this again looks like a struct to me rather than a class...

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+      nas_5g_auth_info_proc_t* auth_info_proc);
+  int amf_send_authentication_request(
+      amf_context_t* amf_context, nas_amf_auth_proc_t* auth_proc);
+};
```
does not make much sense to me, why inherit amf_context? why group them in a class? just place all these as utility functions in a fiveg authentication procedure header.

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,1001 @@
+      amf_nas_message_decode_status_t* decode_status);
+  nas_amf_common_proc_t* get_nas5g_common_procedure(
+      const amf_context_t* const ctxt, amf_common_proc_type_t proc_type);
+};
```
does not make much sense to me why all these are in a class...

chandra-77 replied:
fixed

ulaskozat reviewed (changes requested):
@chandra-77 a lot of class definitions do not make sense to me at all. They are either collection of methods with no specific ownership of data or pure structs with no intended methods.

ssanadhya commented:
@chandra-77 , could you please confirm that S1ap integration tests are passing with this PR when the converged_core flag is set to false?

chandra-77 commented:
> @chandra-77 , could you please confirm that S1ap integration tests are passing with this PR when the converged_core flag is set to false?

@ssanadhya , we did not test the S1AP tests because this PR have dependency on other PRs, I will set converged_core  as false and run the S1ap integ_tests. I will let you know once I ran tests.

ssanadhya commented on lte/gateway/c/oai/tasks/amf/CMakeLists.txt:
```diff
@@ -0,0 +1,81 @@
+set(NGAP_C_DIR ${PROJECT_BINARY_DIR}/ngap/r16.0)
+set(NAS5G_C_DIR /home/vagrant/magma/lte/gateway/c/oai/tasks/nas5g)
```
```suggestion
set(NAS5G_C_DIR ${PROJECT_SOURCE_DIR}/tasks/nas5g)
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/CMakeLists.txt:
```diff
@@ -0,0 +1,81 @@
+set(NGAP_C_DIR ${PROJECT_BINARY_DIR}/ngap/r16.0)
+set(NAS5G_C_DIR /home/vagrant/magma/lte/gateway/c/oai/tasks/nas5g)
+set(NAS_C_DIR /home/vagrant/magma/lte/gateway/c/oai)
```
No need to define this variable, just use `${PROJECT_SOURCE_DIR}` wherever you need to refer to `/home/vagrant/magma/lte/gateway/c/oai`

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/CMakeLists.txt:
```diff
@@ -0,0 +1,81 @@
+set(NAS5G_C_DIR /home/vagrant/magma/lte/gateway/c/oai/tasks/nas5g)
+set(NAS_C_DIR /home/vagrant/magma/lte/gateway/c/oai)
+set(NAS_DIR /home/vagrant/magma/lte/gateway/c/oai/tasks/nas)
+set(AMFAPP_STATE_CPP_PROTOS mme_nas_state nas_state common_types std_3gpp_types)
```
Not needed here as you are not persisting any task state to Redis

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/CMakeLists.txt:
```diff
@@ -0,0 +1,81 @@
+generate_grpc_protos("${S6ASRV_FEG_GRPC_PROTOS}" "${PROTO_SRCS}"
+  "${PROTO_HDRS}" ${FEG_PROTO_DIR} ${FEG_OUT_DIR})
+
+set(LTE_OUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/lte/protos/oai")
```
Unused, please remove

chandra-77 replied:
removed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_main.cpp:
```diff
@@ -0,0 +1,128 @@
+task_zmq_ctx_t amf_app_task_zmq_ctx;
+void amf_app_exit(void);
+static int handle_message(zloop_t* loop, zsock_t* reader, void* arg) {
+  zframe_t* msg_frame = zframe_recv(reader);
```
Please use the `receive_msg` function added recently (https://github.com/magma/magma/pull/5262/files#diff-0127e0adc6ca4891c8ec0d0ed0f4cc28715b4f7f894c917bbef5faab9ed1965c)  to avoid unaligned memory access.

chandra-77 replied:
as we are working on little bit old master we cant use receive_msg right now, we will update this PR with fix in next commit.

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_main.cpp:
```diff
@@ -0,0 +1,128 @@
+      amf_app_handle_pdu_session_response(
+          &N11_CREATE_PDU_SESSION_RESPONSE(received_message_p));
+      break;
+    case NGAP_PDUSESSIONRESOURCE_SETUP_RSP:
```
@sanjay-ACL @sureshgorijavolu can we have a consistent naming convention here? This can be fixed as part of a separate PR.

```suggestion
    case NGAP_PDU_SESSION_RESOURCE_SETUP_RSP:
```

chandra-77 replied:
will fix this in separate PR

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_main.cpp:
```diff
@@ -0,0 +1,128 @@
+    case TERMINATE_MESSAGE:
+      OAILOG_DEBUG(LOG_AMF_APP, "TERMINATE_MESSAGE received\n");
+      itti_free_msg_content(received_message_p);
+      zframe_destroy(&msg_frame);
```
Please refer to changes in https://github.com/magma/magma/pull/5262/files#diff-0127e0adc6ca4891c8ec0d0ed0f4cc28715b4f7f894c917bbef5faab9ed1965c and replace this with `free`

chandra-77 replied:
as we are working on little bit old master, we will update this PR with the fix in next commit.

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_config.c:
```diff
@@ -0,0 +1,144 @@
+**                                                                        **
+**                                                                        **
+***************************************************************************/
+int amf_config_parse_opt_line(int argc, char* argv[], amf_config_t* config_pP) {
```
What's the purpose of this function if it is only calling `amf_config_init()`?

ssanadhya replied:
Is it still needed after the changes in https://github.com/magma/magma/pull/5248/files#diff-42ceeb495dca630b224ba4192bab577b40ddfb0586fe01426dcbeab8498d5d46 ?

chandra-77 replied:
cross verified with ngap PRs, will remove this function in upcoming PR

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_handler.cpp:
```diff
@@ -0,0 +1,628 @@
+//------------------------------------------------------------------------------
+ue_m5gmm_context_s* amf_ue_context_exists_guti(
+    amf_ue_context_t* const amf_ue_context_p, const guti_m5_t* const guti_p) {
+  hashtable_rc_t h_rc       = HASH_TABLE_OK;
+  uint64_t amf_ue_ngap_id64 = 0;
+  h_rc                      = obj_hashtable_uint64_ts_get(
+      amf_ue_context_p->guti_ue_context_htbl, (const void*) guti_p,
+      sizeof(*guti_p), &amf_ue_ngap_id64);
+
+  if (HASH_TABLE_OK == h_rc) {
+    // TODO: this method is deprecated and will be removed once the AMF's
+    // context is migrated to map in the upcoming multi-UE PR
+  } else {
+    OAILOG_WARNING(LOG_AMF_APP, " No GUTI hashtable for GUTI ");
+  }
+
```
Can we just clean this function and always return NULL here for clarity? 
Retain the TODO comment

chandra-77 replied:
Hash tables will replace with MAPs, this function will be removed in upcoming PR.

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_authentication.cpp:
```diff
@@ -0,0 +1,595 @@
+ **                                                                        **
+ ** Name:        nas_itti_auth_info_req()                                  **
+ **                                                                        **
+ ** Description: Sends Authenticatio Req to UDM via S6a Task               **
```
Spell check

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_authentication.cpp:
```diff
@@ -0,0 +1,595 @@
+     */
+    // void* timer_callback_args = NULL;
+    // nas_stop_T3560(ue_mm_context->amf_ue_ngap_id, &auth_proc->T3560,
+    // timer_callback_args);
```
Remove commented code

chandra-77 replied:
removed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,702 @@
+#define AMF_APP_UE_CONTEXT_MODIFICATION_TIMER_VALUE 2       // In seconds
+#define AMF_APP_PAGING_RESPONSE_TIMER_VALUE 4               // In seconds
+#define AMF_APP_ULR_RESPONSE_TIMER_VALUE 3                  // In seconds
+#define NAS_SECURITY_ALGORITHMS_MINIMUM_LENGTH 1
```
```suggestion
#define NAS5G_SECURITY_ALGORITHMS_MINIMUM_LENGTH 1
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context.cpp:
```diff
@@ -0,0 +1,408 @@
+#include "nas5g_network.h"
+#include "amf_app_state_manager.h"
+
+using namespace std;
```
Avoid the `using namespace` directive, instead use the `std::` prefix wherever needed

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context.cpp:
```diff
@@ -0,0 +1,408 @@
+extern task_zmq_ctx_t amf_app_task_zmq_ctx;
+ue_m5gmm_context_s ue_m5gmm_global_context;  // TODO : This has been taken care
+                                             // in new PR with multi UE feature
+static void _directoryd_report_location(uint64_t imsi, uint8_t imsi_len) {
```
Remove the underscore prefix

```suggestion
static void directoryd_report_location(uint64_t imsi, uint8_t imsi_len) {

```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context.cpp:
```diff
@@ -0,0 +1,408 @@
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    notify_ngap_new_ue_amf_ngap_id_association()                  **
```
Use the correct function name

chandra-77 replied:
fixed

ssanadhya reviewed (changes requested):
First parse on the PR, will continue reviewing

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_handler.cpp:
```diff
@@ -104,12 +97,22 @@ void amf_ue_context_update_coll_keys(
+        OAILOG_ERROR(
+            LOG_AMF_APP,
+            "Insertion of Hash entry failed for  amf_ue_ngap_id "
+            "0x%08" PRIX64 " \n",
```
Please use `AMF_UE_NGAP_ID_FMT` to specify the print formatting of amf_ue_ngap_id

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_handler.cpp:
```diff
@@ -104,12 +97,22 @@ void amf_ue_context_update_coll_keys(
+    OAILOG_ERROR(
+        LOG_AMF_APP,
+        "Invalid  amf_ue_ngap_id "
+        "0x%08" PRIX64 " \n",
```
Please use `AMF_UE_NGAP_ID_FMT`

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/CMakeLists.txt:
```diff
@@ -0,0 +1,75 @@
+set(NGAP_C_DIR ${PROJECT_BINARY_DIR}/ngap/r16.0)
+set(NAS5G_C_DIR ${PROJECT_SOURCE_DIR}/tasks/nas5g)
+set(NAS_C_DIR ${PROJECT_SOURCE_DIR})
```
No need to copy one var into another, you can just use ${PROJECT_SOURCE_DIR} wherever you need NAS_C_DIR

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context.cpp:
```diff
@@ -0,0 +1,414 @@
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    notify_ngap_new_ue_amf_ngap_id_association()                  **
```
Fix function name

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,714 @@
+
+// Header length boundaries of 5GS Mobility Management messages
+#define AMF_HEADER_LENGTH sizeof(amf_msg_header)
+#define AMF_HEADER_MAXIMUM_LENGTH AMF_HEADER_LENGTH
```
What's the purpose of this duplicate definition? We can just use one of these.

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context.cpp:
```diff
@@ -0,0 +1,414 @@
+      AMF_APP_TIMER_INACTIVE_ID, AMF_APP_INITIAL_CONTEXT_SETUP_RSP_TIMER_VALUE};
+  new_p->m5_paging_response_timer = (amf_app_timer_t){
+      AMF_APP_TIMER_INACTIVE_ID, AMF_APP_PAGING_RESPONSE_TIMER_VALUE};
+  new_p->m5_ulr_response_timer = (amf_app_timer_t){
+      AMF_APP_TIMER_INACTIVE_ID, AMF_APP_ULR_RESPONSE_TIMER_VALUE};
+  new_p->m5_ue_context_modification_timer = (amf_app_timer_t){
+      AMF_APP_TIMER_INACTIVE_ID, AMF_APP_UE_CONTEXT_MODIFICATION_TIMER_VALUE};
```
Are either of these getting used in this or future PRs? If not, please remove

chandra-77 replied:
getting used in upcoming PRs

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context.cpp:
```diff
@@ -0,0 +1,414 @@
+ ***************************************************************************/
+// warning: lock the UE context
+ue_m5gmm_context_s* amf_create_new_ue_context(void) {
+  ue_m5gmm_context_s* new_p = new (ue_m5gmm_context_s);
```
Can you confirm if this is indeed zero-initialized? What is the behavior of `new` for nested structures?

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,714 @@
+/* Operation on UE context structure
+ * TODO need to add more data members and member functions in future
+ */
+class amf_app_ue_context {
```
What's the purpose of this class with only member functions? 

Further, why is `amf_insert_ue_context` a member of this but not `amf_create_new_ue_context`?

chandra-77 replied:
removed the class

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,714 @@
+
+ue_m5gmm_context_s* amf_create_new_ue_context(void);
+
+// Retrive required UE context from respective hash table
```
Spell check

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,714 @@
+  int amf_msg_encode_header(
+      const amf_msg_header* header, uint8_t* buffer, uint32_t len);
+  int amf_msg_decode(AMFMsg* msg, uint8_t* buffer, uint32_t len);
+  int AmfMsgEncode(AMFMsg* msg, uint8_t* buffer, uint32_t len);
```
Please follow consistent naming conventions.

chandra-77 replied:
naming fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,714 @@
+  int amf_msg_encode_header(
+      const amf_msg_header* header, uint8_t* buffer, uint32_t len);
+  int amf_msg_decode(AMFMsg* msg, uint8_t* buffer, uint32_t len);
+  int AmfMsgEncode(AMFMsg* msg, uint8_t* buffer, uint32_t len);
```
Declared but undefined in this PR.

chandra-77 replied:
resolved

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -0,0 +1,714 @@
+
+typedef struct amf_registration_request_ies_s {
+  amf_proc_registration_type_t
+      m5gsregistrationtype;  // m5gsregistrationtype=AMF_REGISTRATION_TYPE_RESERVED;
```
Remove commented code

chandra-77 replied:
removed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_as.h:
```diff
@@ -0,0 +1,72 @@
+#include "amf_asDefs.h"
+#include "amf_as_message.h"
+
+using namespace std;
```
Avoid `using namespace` directive, follow the guidelines here https://google.github.io/styleguide/cppguide.html#Namespaces and use the prefix 'std::' wherever needed

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_asDefs.h:
```diff
@@ -0,0 +1,174 @@
+  guti_m5_t* guti;              // GUTI temporary mobile identity
+  amf_as_security_data_t sctx;  // M5G NAS security context
+#define AMF_AS_NAS_DATA_REGISTRATION_ACCEPT 0x04    // REGISTRATION Accept
+#define AMF_AS_NAS_AMF_INFORMATION 0x05             // Emm information
```
```suggestion
#define AMF_AS_NAS_EMM_INFORMATION 0x05             // Emm information
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_asDefs.h:
```diff
@@ -0,0 +1,174 @@
+  tai_t tai;          // The first tracking area the UE is registered
+  ecgi_t ecgi;  // E-UTRAN CGI This information element is used to globally
+                // identify a cell
+#define AMF_AS_NAS_INFO_REGISTERD 0x01  // REGISTERD request
```
Spell check

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_authentication.h:
```diff
@@ -0,0 +1,62 @@
+                // due to sync failure
+} nas5g_auth_info_proc_t;
+
+nas5g_auth_info_proc_t* nas_new_5gcn_auth_info_procedure(
```
No need for the `new` qualifier, there is only one nas5g auth info procedure
```suggestion
nas5g_auth_info_proc_t* nas5g_cn_auth_info_procedure(
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_data.h:
```diff
@@ -0,0 +1,97 @@
+#endif
+#include "amf_securityDef.h"
+
+using namespace std;
```
Why is this needed here?

chandra-77 replied:
resolved

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_data.h:
```diff
@@ -0,0 +1,97 @@
+
+// AMF security context
+typedef struct amf_security_context_s {
+  amf_sc_type_t sc_type; /* Type of security context        */
```
Where is this type getting defined?

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_data.h:
```diff
@@ -0,0 +1,97 @@
+// AMF security context
+typedef struct amf_security_context_s {
+  amf_sc_type_t sc_type; /* Type of security context        */
+  /* state of security context is implicit due to its storage location
```
What's the purpose of this comment?

chandra-77 replied:
removed the comment

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_data.cpp:
```diff
@@ -0,0 +1,115 @@
+#if DEBUG_IS_ON
+  char imsi_str[IMSI_BCD_DIGITS_MAX + 1] = {0};
+  IMSI64_TO_STRING(ctxt->_imsi64, imsi_str, ctxt->_imsi.length);
+#endif
```
What's the purpose of these lines if there is no LOG message using the `imsi_str`?

chandra-77 replied:
LOG added

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_identity.cpp:
```diff
@@ -0,0 +1,193 @@
+/**
+ *copyright 2020 The Magma Authors.
```
```suggestion
 * Copyright 2020 The Magma Authors.
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_transport.cpp:
```diff
@@ -0,0 +1,80 @@
+#include "nas5g_network.h"
+#include "amf_app_state_manager.h"
+
+using namespace std;
```
Is this getting used anywhere? See other comments on `using namespace`

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_transport.cpp:
```diff
@@ -0,0 +1,80 @@
+   * Store the NGAP NAS DL DATA REQ in case of IMSI or combined 5GMM/IMSI
+   * deregister in sgs context and send it after recieving the 5GS IMSI
+   * deregister Ack from 5GS task.
+   */
```
What's the purpose of this comment?

chandra-77 replied:
removed the comment

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_state_manager.h:
```diff
@@ -0,0 +1,98 @@
+ */
+amf_app_desc_t* get_amf_nas_state(bool read_from_redis);
+
+// Retriving respective global hash table
```
Spell check

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_config.c:
```diff
@@ -0,0 +1,146 @@
+  config->max_ues                        = 2;
+  config->unauthenticated_imsi_supported = 0;
+  config->relative_capacity              = RELATIVE_CAPACITY;
+  config->amf_statistic_timer            = AMF_STATISTIC_TIMER_S;
```
Also initialize `use_stateless` to false since this code does not write to Redis at the moment

`config->use_stateless = false;`

chandra-77 replied:
resolved

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_identity.cpp:
```diff
@@ -0,0 +1,193 @@
+      amf_ctx->_imsi.length = 8;
+      amf_ctx->_m5_guti     = *amf_ctx_guti;
+    } else {
+      AssertFatal(
```
Use `AssertFatal` with caution. This one can be replaced with an ERROR log, since the entire service should not crash if one UE sends a bad Identity Response.

chandra-77 replied:
replaced with ERROR log

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_identity.h:
```diff
@@ -0,0 +1,90 @@
+#include "amf_app_ue_context_and_proc.h"
+#include "M5GIdentityResponse.h"
+
+using namespace std;
```
Remove if not needed or replace with usage of `std::`

chandra-77 replied:
resolved

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_recv.h:
```diff
@@ -0,0 +1,63 @@
+int amf_handle_authentication_response(
+    amf_ue_ngap_id_t ue_id, AuthenticationResponseMsg* msg, int amf_cause,
+    amf_nas_message_decode_status_t status);
+int amf_handle_securitycomplete_response(
```
```suggestion
int amf_handle_security_complete_response(
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_recv.h:
```diff
@@ -0,0 +1,63 @@
+    amf_nas_message_decode_status_t status);
+int amf_handle_securitycomplete_response(
+    amf_ue_ngap_id_t ue_id, amf_nas_message_decode_status_t decode_status);
+int amf_handle_registrationcomplete_response(
```
```suggestion
int amf_handle_registration_complete_response(
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_sap.cpp:
```diff
@@ -0,0 +1,142 @@
+    data->is_new = is_new;
+    data->ksi    = context->eksi;
+    data->sqn    = context->dl_count
+                    .seq_num;  // TODO AT check whether we need to
```
Is this TODO addressed in future PR?

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_sap.h:
```diff
@@ -0,0 +1,152 @@
+  AMFREG_COMMON_PROC_REJ,
+  AMFREG_COMMON_PROC_ABORT,
+  /* AMFAS-SAP */
+  AMFAS_SECURITY_REQ        = _AMFAS_SECURITY_REQ,
+  AMFAS_SECURITY_IND        = _AMFAS_SECURITY_IND,
+  AMFAS_SECURITY_RES        = _AMFAS_SECURITY_RES,
+  AMFAS_SECURITY_REJ        = _AMFAS_SECURITY_REJ,
+  AMFAS_ESTABLISH_REQ       = _AMFAS_ESTABLISH_REQ,
+  AMFAS_ESTABLISH_CNF       = _AMFAS_ESTABLISH_CNF,
+  AMFAS_ESTABLISH_REJ       = _AMFAS_ESTABLISH_REJ,
+  AMFAS_RELEASE_REQ         = _AMFAS_RELEASE_REQ,
+  AMFREG_REGISTRATION_REJ   = _AMFREG_REGISTRATION_REJ,
+  AMFAS_DATA_IND            = _AMFAS_DATA_IND,
+  AMFREG_REGISTRATION_CNF   = _AMFREG_REGISTRATION_CNF,
+  AMFREG_DEREGISTRATION_REQ = _AMFREG_DEREGISTRATION_REQ,
+  AMFAS_DATA_REQ            = _AMFAS_DATA_REQ,
```
What's the purpose of duplicating these enums?

chandra-77 replied:
We leveraged these enums from LTE.

ssanadhya commented on lte/gateway/c/oai/tasks/amf/nas5g_network.h:
```diff
@@ -0,0 +1,32 @@
+};
+#endif
+
+using namespace std;
```
Remove

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/nas5g_network.h:
```diff
@@ -0,0 +1,32 @@
+void bdestroy_wrapper(bstring* b);
+
+// Free up ptr
+void free_wrapper(void** ptr);
```
Why are these getting duplicated? Can we not use the free_wrapper defined in `dynamic_memory_check.c` ?

chandra-77 replied:
resolved

ssanadhya commented on lte/gateway/c/oai/tasks/amf/nas5g_network.h:
```diff
@@ -0,0 +1,32 @@
+using namespace std;
+namespace magma5g {
+// Free up the bstring
+void bdestroy_wrapper(bstring* b);
```
Why is this getting duplicated?

chandra-77 replied:
resolved

ssanadhya commented on lte/gateway/c/oai/tasks/amf/nas_network.cpp:
```diff
@@ -0,0 +1,43 @@
+#endif
+#include "nas5g_network.h"
+
+using namespace std;
```
Unused, please remove

chandra-77 replied:
resolved

ssanadhya reviewed (changes requested):
Set 2 of review comments. Will add more comments in next round

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_defs.h:
```diff
@@ -0,0 +1,42 @@
+#include "n11_messages_types.h"
+#include "amf_app_ue_context_and_proc.h"
+
+using namespace std;
```
Remove

chandra-77 replied:
Removed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/prepare_request_for_smf.cpp:
```diff
@@ -0,0 +1,159 @@
+using magma::lte::SetSMSessionContext;
+using magma5g::AsyncSmfServiceClient;
+
+namespace {
```
What's the purpose of an unnamed namespace here?

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/prepare_request_for_smf.cpp:
```diff
@@ -0,0 +1,159 @@
+using magma5g::AsyncSmfServiceClient;
+
+namespace {
+using namespace magma::lte;
```
Remove this directive and use the magma::lte:: prefix where needed, or explicitly list the class being used from magma::lte:: at the top.

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,849 @@
+        "amf_procedures\n");
+    amf_context->amf_procedures = _nas_new_amf_procedures(amf_context);
+  }
+  amf_context->amf_procedures->amf_specific_proc = new nas_amf_specific_proc_t;
```
Can you confirm that `new` is initializing this struct to zero, since it is nested structure?

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,849 @@
+      get_nas_specific_procedure_registration(amf_context);
+
+  if (registration_proc) {
+    rc = RETURNok;
```
Remove

chandra-77 replied:
Removed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,849 @@
+ **      Others:    None                                                   **
+ **                                                                        **
+ ***************************************************************************/
+int amf_handle_registrationcomplete_response(
```
```suggestion
int amf_handle_registration_complete_response(
```

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,849 @@
+      /*currently by default Activate  Bearer Context Accept message was
+       * sent in Registration complete Now, modified the code to send the
+       * message received in Uplink/smfContainer. third byte of smf message
+       * container is a message_type*/
```
What is the purpose of this comment?

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,849 @@
+  }
+
+  rc = RETURNok;
+  if ((rc != RETURNerror)) {
```
What's the purpose of this condition if rc is always set to RETURNok?

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,849 @@
+        "message has been discarded for"
+        "ue_id = (%u)\n",
+        ue_id);
+    rc = RETURNok;
```
Which SMF procedure is being referred to here?

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,849 @@
+ **                                                                        **
+ ***************************************************************************/
+
+int amf_proc_amf_informtion(ue_m5gmm_context_s* ue_amf_ctx) {
```
```suggestion
int amf_proc_amf_information(ue_m5gmm_context_s* ue_amf_ctx) {
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,849 @@
+  amf_as->amf_as_set_security_data(
+      &amf_as->sctx, &amf_ctx->_security, false, true);
+  /*
+   * Notify AMF-AS SAP that TAU Accept message has to be sent to the network
```
```suggestion
   * Notify AMF-AS SAP that Registration Accept message has to be sent to the network
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -0,0 +1,849 @@
+  OAILOG_FUNC_IN(LOG_NAS_AMF);
+  int rc = RETURNok;
+  OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
+}
```
What's the purpose of this function definition? Is this getting populated in a future PR?

chandra-77 replied:
Will be populated in upcoming PRs

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp:
```diff
@@ -0,0 +1,99 @@
+ **                                                                        **
+ **                                                                        **
+ ***************************************************************************/
+int amf_handle_securitycomplete_response(
```
```suggestion
int amf_handle_security_complete_response(
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp:
```diff
@@ -0,0 +1,99 @@
+      rc = amf_sap_send(&amf_sap);
+    }
+    /* Nothing to do in
+     * Calling SMC response success and triggering registration accept message*/
```
Fix comment

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp:
```diff
@@ -0,0 +1,99 @@
+  if (ue_mm_context) {
+    amf_ctx = &ue_mm_context->amf_context;
+  } else {
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, RETURNerror);
```
Add error log here

chandra-77 replied:
Log added

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_handler.cpp:
```diff
@@ -0,0 +1,677 @@
+      ue_context_p->amf_context._imsi64,
+      GUTI_ARG_M5G(&ue_context_p->amf_context._guti));
+
+  if ((INVALID_GNB_UE_NGAP_ID_KEY != gnb_ngap_id_key) &&
```
```suggestion
  if ((gnb_ngap_id_key != INVALID_GNB_UE_NGAP_ID_KEY ) &&
```

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_handler.cpp:
```diff
@@ -0,0 +1,677 @@
+    ue_context_p->gnb_ngap_id_key = gnb_ngap_id_key;
+  }
+
+  if (INVALID_AMF_UE_NGAP_ID != amf_ue_ngap_id) {
```
Please put macro on right hand side of operator

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_msg.cpp:
```diff
@@ -0,0 +1,59 @@
+      itti_alloc_new_message(TASK_AMF_APP, NGAP_UE_CONTEXT_RELEASE_COMMAND);
+
+  if (message_p == NULL) {
+    OAILOG_FUNC_OUT(LOG_AMF_APP);
```
Log an error here

chandra-77 replied:
Log added

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_msg.cpp:
```diff
@@ -0,0 +1,59 @@
+  if (message_p == NULL) {
+    OAILOG_FUNC_OUT(LOG_AMF_APP);
+  }
+
```
Memset the NGAP_UE_CONTEXT_RELEASE_COMMAND(message_p) to zeros

chandra-77 replied:
Resolved

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,247 @@
+        // many more to be coded
+      default:
+        *dl_pdu_ambr = (256 * smf_context->dl_session_ambr);
+        break;
```
Isn't this simply (4^(smf_context->dl_ambr_unit-1)) * smf_context->dl_session_ambr for values 1-4 ?

chandra-77 replied:
Replaced with suggested math equation

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,247 @@
+      case 4:
+        *ul_pdu_ambr = (64 * smf_context->ul_session_ambr);
+        break;
+        // many more to TBD
```
This can be compressed into `(4^(smf_context->ul_ambr_unit-1)) * smf_context->ul_session_ambr `

chandra-77 replied:
Replaced with suggested math equation

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -0,0 +1,247 @@
+          .qos_flow_setup_request_list.qos_flow_req_item
+          .qos_flow_level_qos_param.alloc_reten_priority.pre_emption_cap;
+
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
```
Very verbose! Can be abstracted into a function/macro which takes as arguments 
- `amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item`
- `smf_context->pdu_resource_setup_req.pdu_session_resource_setup_request_transfer.qos_flow_setup_request_list.qos_flow_req_item`

chandra-77 replied:
Fixed with memcopy(), LHS and RHS have same structure

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context.cpp:
```diff
@@ -0,0 +1,414 @@
+ **                                                                        **
+ ** Description: Checks if UE context exists already or not                **
+ **                                                                        **
+ **                                                                        **
```
Fixe description

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_ue_context.cpp:
```diff
@@ -0,0 +1,414 @@
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    notify_ngap_new_ue_amf_ngap_id_association()                  **
```
Fix function name

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_security_mode_control.cpp:
```diff
@@ -0,0 +1,300 @@
+  if (smc_proc) {
+    /*
+     * Setup ongoing AMF procedure callback functions
+     */
```
Single line comment

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_security_mode_control.cpp:
```diff
@@ -0,0 +1,300 @@
+    smc_proc->amf_com_proc.amf_proc.base_proc.fail_out      = NULL;
+    /*
+     * Set the UE identifier
+     */
```
Remove

chandra-77 replied:
Removed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_security_mode_control.cpp:
```diff
@@ -0,0 +1,300 @@
+    smc_proc->ue_id = ue_id;
+    /*
+     * Reset the retransmission counter
+     */
```
Remove

chandra-77 replied:
Removed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_security_mode_control.cpp:
```diff
@@ -0,0 +1,300 @@
+    smc_proc->retransmission_count = 0;
+    /*
+     * Set the key set identifier
+     */
```
Remove all bloated comments, these are unnecessary for simple statements

chandra-77 replied:
Taken care

ssanadhya commented on lte/gateway/c/oai/tasks/amf/deregistration_request.cpp:
```diff
@@ -0,0 +1,300 @@
+  }
+  /*value of access_type would be 1 or 2 or 3, 24-501 - 9.11.3.20 */
+  switch (msg->m5gs_de_reg_type.access_type) {
+    case 1:
```
Use macro `AMF_3GPP_ACCESS` instead of the numeric value 
```suggestion
    case AMF_3GPP_ACCESS:
```

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/deregistration_request.cpp:
```diff
@@ -0,0 +1,300 @@
+          "Access type AMF_NONE_3GPP_ACCESS for deregistration request from "
+          "UE\n");
+      break;
+    case 3:
```
Replace the numeric constants with macros

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/deregistration_request.cpp:
```diff
@@ -0,0 +1,300 @@
+ *        Sends ue context release notification to NGAP
+ *        Clean up availble ue context
+ *        send message to SMF to remove all PDU sessions
+ *        AMF is not expecting response from SMF.
```
Split this as comments inside the Function itself

chandra-77 replied:
Resolved

ssanadhya commented on lte/gateway/c/oai/tasks/amf/deregistration_request.cpp:
```diff
@@ -0,0 +1,300 @@
+  if (!ue_context) {
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
+  }
+  // TODO Condition for teid of N11, active_pdn_contexts and amy more
```
Fix comment

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/deregistration_request.cpp:
```diff
@@ -0,0 +1,300 @@
+    OAILOG_FUNC_OUT(LOG_AMF_APP);
+  }
+
+  /* UE state had not been mainted & nothing to remove
```
spell check

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/deregistration_request.cpp:
```diff
@@ -0,0 +1,300 @@
+          " gnb_ue_ngap_id"
+          "0x%08" PRIX64
+          " amf_ue_ngap_id "
+          "0x%08" PRIX64 " not in IMSI collection\n",
```
Use the formatting macros

chandra-77 replied:
Replaced with macros

ssanadhya commented on lte/gateway/c/oai/tasks/amf/nas_proc.cpp:
```diff
@@ -0,0 +1,303 @@
+    const identity_type2_t type, success_cb_t success, failure_cb_t failure) {
+  OAILOG_FUNC_IN(LOG_NAS_AMF);
+  int rc                     = RETURNerror;
+  amf_context->amf_fsm_state = AMF_REGISTERED;
```
Why is the FSM state set to REGISTERED here?

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_recv.cpp:
```diff
@@ -0,0 +1,292 @@
+  /*
+   * Message checking
+   */
+  if (NULL == msg->autn_response_parameter.response_parameter) {
```
Put macro on the right hand side
```suggestion
  if (msg->autn_response_parameter.response_parameter == NULL) {
```

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_as.cpp:
```diff
@@ -0,0 +1,1003 @@
+ **              Others:        None                                       **
+ **                                                                        **
+ ***************************************************************************/
+static int amf_dl_nas_transportmsg(
```
```suggestion
static int amf_dl_nas_transport_msg(
```

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_as.cpp:
```diff
@@ -0,0 +1,1003 @@
+  /*
+   * A valid 5G CN security context exists but NAS integrity key
+   * * * * is not available
+   */
```
What's the purpose of this comment?

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_as.cpp:
```diff
@@ -0,0 +1,1003 @@
+ ** Return:   size                                                        **
+ **                                                                       **
+ **************************************************************************/
+static int amf_send_identity_request(
```
This function is not sending the identity request message. Is the plan to add it in future PR?

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_as.cpp:
```diff
@@ -0,0 +1,1003 @@
+        "status:%d, StatusCode:%d\n",
+        status.error_message(), response.error_code());
+  }
+}
```
Why is the S6a function here? Can we not use it from the existing S6a client api?

chandra-77 replied:
Will be taken care in upcoming PR

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_as.cpp:
```diff
@@ -0,0 +1,1003 @@
+
+/***************************************************************************
+ **                                                                       **
+ ** Name:        amf_sned_auth_request()                                  **
```
Spell check

chandra-77 replied:
fixed

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_asDefs.h:
```diff
@@ -0,0 +1,173 @@
+} amf_as_primitive_t;
+
+typedef enum amfcn_primitive_s {
+  _AMFCN_START = 400,
```
remove _ at the beginning

chandra-77 replied:
We leveraged these enums from LTE code, we followed the same naming.

ulaskozat commented on lte/gateway/c/oai/tasks/amf/amf_asDefs.h:
```diff
@@ -0,0 +1,173 @@
+  AMFCN_DEACTIVATE_PDUSESSION_REQ,
+  AMFCN_IDENTITY_PARAM_RES,
+  AMFCN_SMC_PARAM_RES,
+  _AMFCN_END
```
remove _ at the beginning. Please check any usage of `_` or `__` at the beginning of variables, object or function names and remove them.

chandra-77 replied:
Those enums are leveraged from LTE, we followed the same naming. Remaining variables, objects and functions names are taken care.

Leveraged enums are used in many places if we need to remove the _ at the begining of those, we will address them in separate PR.

ulaskozat reviewed (approved):
LGTM with couple of nit comments on the use of `_`.

ulaskozat commented:
@chandra-77 please update the summary as you have to run the s1ap test cases which has no dependency on this PR.

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -571,25 +570,23 @@ int amf_send_registration_accept(amf_context_t* amf_context) {
+      if (RETURNerror == rc) {
         OAILOG_DEBUG(
-            LOG_NAS_AMF, " AMF-PROC  - Sending  Registration Accept\n");
+            LOG_NAS_AMF, " AMF-PROC  - Sending to access point failed\n");
```
What is the access point here?

ssanadhya commented on lte/gateway/c/oai/tasks/amf/Registration.cpp:
```diff
@@ -741,52 +733,33 @@ int amf_proc_registration_complete(
-    }
-  } else {
+  /*
+   * Set the network registrationment indicator
```
```suggestion
   * Set the network registration indicator
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_Security_Mode.cpp:
```diff
@@ -61,12 +60,13 @@ int amf_handle_securitycomplete_response(
   if (smc_proc) {
     /*
-     * TODO:Stop timer T3570 This is taken care with Timer Implementation PR
+     * TODO:Stop timer T3570 This to be taken care n upcoming PR
```
```suggestion
     * TODO:Stop timer T3570 This to be taken care in upcoming PR
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_handler.cpp:
```diff
@@ -99,8 +98,8 @@ void amf_ue_context_update_coll_keys(
-            "Insertion of Hash entry failed for  amf_ue_ngap_id "
-            "0x%08" PRIX64 " \n",
+            "Insertion of Hash entry failed for  "
+            "amf_ue_ngap_id " AMF_UE_NGAP_ID_FMT PRIX32 " \n",
```
Why do we need PRIX32 here? Please absorb it in the definition of AMF_UE_NGAP_ID_FMT

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_handler.cpp:
```diff
@@ -425,6 +419,7 @@ imsi64_t amf_app_handle_initial_ue_message(
         initial_pP->ecgi, initial_pP->m5g_rrc_establishment_cause, s_tmsi,
         initial_pP->nas);
   }
+  return 0;
```
```suggestion
  return RETURNok;
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_main.cpp:
```diff
@@ -115,6 +112,7 @@ static int handle_message(zloop_t* loop, zsock_t* reader, void* arg) {
           ITTI_MSG_ID(received_message_p), ITTI_MSG_NAME(received_message_p));
       break;
   }
+  return 0;
```
```suggestion
  return RETURNok;
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -45,42 +44,18 @@ void ambr_calculation_pdu_session(
-        break;
+    if ((smf_context->dl_ambr_unit) < 4) {
+      *dl_pdu_ambr =
+          4 ^ (smf_context->dl_ambr_unit) * (smf_context->dl_session_ambr);
```
Check the math here
```suggestion
          4 ^ (smf_context->dl_ambr_unit - 1) * (smf_context->dl_session_ambr);
```

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp:
```diff
@@ -45,42 +44,18 @@ void ambr_calculation_pdu_session(
-        *ul_pdu_ambr = (256 * smf_context->ul_session_ambr);
-        break;
+    if ((smf_context->ul_ambr_unit) < 4) {
+      *ul_pdu_ambr =
```
Check the math here

chandra-77 replied:
fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/amf_authentication.cpp:
```diff
@@ -554,9 +552,6 @@ int amf_proc_authentication_complete(
-   * TODO, Later need to take care on RAN Vector related LIST
-   * nas5g_auth_info_proc_t = get_nas5g_cn_procedure_auth_info()
-   * will be used to manage RAN vector
    * Invoking success directly to hadle security mode command
```
Spell check

chandra-77 replied:
Fixed

ssanadhya commented on lte/gateway/c/oai/tasks/amf/deregistration_request.cpp:
```diff
@@ -263,9 +246,9 @@ void amf_remove_ue_context(
           "UE context not found for n11_ue_context\n"
           " gnb_ue_ngap_id "
-          "0x%08" PRIX64
+          "0x%08" PRIX32
```
Here and other places in this file, please use GNB_UE_NGAP_ID_FMT and AMF_UE_NGAP_ID_FMT

chandra-77 replied:
fixed

electronjoe committed:
Update files (#5818)

electronjoe committed:
[CI] Turn up Github Action for Shellcheck (#5850)

[ShellCheck - A shell script static analysis tool](https://github.com/koalaman/shellcheck).

Here we use a GH Action provided by the Reviewdog library - which will drop review comments onto the PR with the findings.  These review comments can be "resolved" or commented on.

The mode will presently filter based upon `filter_mode: diff_context` -> meaning only modified lines will be reported for findings.

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

shaddi committed:
[lte][agw] Add S1 Handover preparation message types (#5684)

* Add HOPreparation and HOResourceAlloc message types

Adds message types for HandoverPreparation and
HandoverResourceAllocation stages in the MME.

Signed-off-by: Shaddi Hasan <shasan@fb.com>

shaddi committed:
Fix spelling mistakes in PathSwitchRequest handler (#5807)

Signed-off-by: Shaddi Hasan <shasan@fb.com>

hcgatewood committed:
[docs] Reorganize docusaurus by topic (#5733)

Signed-off-by: Hunter Gatewood <hgatewood@gmail.com>

hcgatewood committed:
[orc8r][subcribers] Add configurable max gRPC msg size (#5432)

Signed-off-by: Hunter Gatewood <hgatewood@gmail.com>

hcgatewood committed:
[orc8r][configurator] Cover last union-find branch (#5632)

Signed-off-by: Hunter Gatewood <hgatewood@gmail.com>

hcgatewood committed:
[docs][orc8r] Fix wrong v1.4 apidocs URL (#5859)

Signed-off-by: Hunter Gatewood <hgatewood@gmail.com>

andreilee committed:
[AGW] Add newline to end of td-agent-bit.conf (#5863)

ssanadhya commented:
@chandra-77 , please rebase onto latest master to resolve the CI checks for `mme_test` and `session_manager_test` .

andreilee committed:
[nms] Add mariadb dependency with yarn (#5869)

Signed-off-by: Andrei Lee <andreilee@fb.com>

hcgatewood committed:
[docs] Add guidelines for force-merge PRs (#5867)

Signed-off-by: Hunter Gatewood <hgatewood@gmail.com>

electronjoe committed:
[CI] Fail gcc-problems checker if build fails (#5851)

Capture the build status for OAI, session_manager when doing GCC warning annotations - return failure status if build fails.

This can then be used as a Required check by Github for a PR Merge.

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

mcallahan committed:
[orc8r] Minor CMakeLists.txt cleanups (#4552)

Co-authored-by: Marie <37634144+themarwhal@users.noreply.github.com>

koolzz committed:
[Pipelined]Report Policy Rule Version to SessionD (#5874)

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

tmdzk committed:
[AGW] fix agw_install.sh problem due to precheck (#5000)

Signed-off-by: Timothée Dzik <timdzik@fb.com>

tmdzk committed:
[DOC]Update doc following agw changes (#5882)

* Update doc following agw changes

Signed-off-by: Timothée Dzik <timdzik@fb.com>

* Update doc to 1.4

Signed-off-by: Timothée Dzik <timdzik@fb.com>

themarwhal committed:
[GITIGNORE] Ignore python build directories generated during packaging (#5883)

Signed-off-by: Marie Bremner <marwhal@fb.com>

koolzz committed:
[MME] Add OVS Flow teid save to register (#5880)

* [MME] Add OVS Flow teid save to register

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

* Set tun id for uplink

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

themarwhal committed:
[MME] Commit uncommitted format changes (#5877)

Signed-off-by: Marie Bremner <marwhal@fb.com>

koolzz committed:
[pipelined]Add Tun ID Register (#5885)

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

uri200 committed:
[subscriberdb][magmad] add print GRPC to other RPC servicers (#5700)

Signed-off-by: Oriol Batalla <obatalla@fb.com>

andreilee committed:
[1.5] Update NMS DB Migration script download reference (#5862)

* [1.5] Update NMS DB Migration script download reference

Signed-off-by: Andrei Lee <andreilee@fb.com>

* [1.5] Update NMS DB Migration script download reference 2

Signed-off-by: Andrei Lee <andreilee@fb.com>

rdefosse committed:
[MME][RHEL][OAI] Better work on building for RHEL8 (#5683)

* [MME][RHEL][OAI] Better work on building for RHEL8, no need for a patch file on a CMakeList.txt file

Signed-off-by: Raphael Defosseux <raphael.defosseux@openairinterface.org>

* [MME][RHEL][OAI] fix on yum update

Signed-off-by: Raphael Defosseux <raphael.defosseux@openairinterface.org>

GANESH-WAVELABS committed:
[Orc8r] 5G WS#2-MVC1 User Guide (#4515)

Signed-off-by: GANESH IRRINKI <ganesh.irrinki@wavelabs.ai>

arunuke committed:
proposal to add resource tagging (#5601)

Signed-off-by: Arun Thulasi <arunt@fb.com>

rsarwad committed:
Added testcase to validate 3495 timer for both default and dedicated bearer while mme restarts (#4016)

* Added testcase to validate for 3495 timer for both default and dedicated bearers

Signed-off-by: rashmi <rashmi.sarwad@radisys.com>

ulaskozat committed:
[lte][agw] Remove hardwired APN APMBR over S1 (#5886)

* Send actual configured APN AMBR to RAN

Signed-off-by: Ulas Kozat <kozat@fb.com>

themarwhal committed:
[AGW] Modify postinst script to append or replace the COMMIT_HASH env var (#5878)

Signed-off-by: Marie Bremner <marwhal@fb.com>

themarwhal committed:
[GITIGNORE] Ignore .venv in git (#5894)

Signed-off-by: GitHub <noreply@github.com>

andreilee committed:
[nms] NMS will use same DB as orc8r for 1.5 (#5858)

Signed-off-by: Andrei Lee <andreilee@fb.com>

ardzoht committed:
[lte][agw] Add hash-based validation to reduce redis writes operations on MME (#5868)

* Redis client write perf improvements

Signed-off-by: Alex Rodriguez <ardzoht@gmail.com>

* Removing state conversion for static PCC rules on SPGW

Signed-off-by: Alex Rodriguez <ardzoht@gmail.com>

* Adding hash based versioning check for redis writes

Signed-off-by: Alex Rodriguez <ardzoht@gmail.com>

pshelar committed:
AGW: GTP driver: improve debuggability of datapath. (#5890)

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

electronjoe committed:
[IWYU] Non-Functional fixup of or8r c includes (#5895)

Signed-off-by: Scott Harrison Moeller <smoeller@fb.com>

Applied IWYU tooling to the `orc8r/gateway/c` sub-targets and corrected all findings (missing and extraneous include headers). This to work towards #4868 which regularly causes build failures as we migrate or update includes / build infrastructure.

Co-authored-by: Scott Harrison Moeller <smoeller@fb.com>

tmdzk committed:
[CI][AGW]Force setuptools version (#5911)

* Force setuptools version

Signed-off-by: Timothée Dzik <timdzik@fb.com>

* Force setuptools version

Signed-off-by: Timothée Dzik <timdzik@fb.com>

ulaskozat committed:
Bug fix in handling ULA message (#5905)

Signed-off-by: Ulas Kozat <kozat@fb.com>

ulaskozat committed:
Eliminate some of the unnecessary state syncs on s1ap (#5888)

Signed-off-by: Ulas Kozat <kozat@fb.com>

pshelar committed:
AGW: OVS: handle L3 port to L3 port packet transfer. (#5889)

OVS action transation code throws asserts on packet that traverse
from L3 to L3 port. This is due to incorrect assumpltion of packet
state. Following patch removes incurrect assert.
This PR also fixes build-ovs.sh by moving OVS patches to 2.8.10
dir.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

tmdzk committed:
Revert changes to push to new registry (#5915)

Signed-off-by: Timothée Dzik <timdzik@fb.com>

themarwhal committed:
[SessionD] Update rule versions on install/uninstalling rules (#5787)

uri200 committed:
[feg] Add ability to use orc8r with S6a_cli (#5677)

themarwhal committed:
[AGW] Commit leftover formatting changes on master (#5920)

Signed-off-by: Marie Bremner <marwhal@fb.com>

andreilee committed:
Update NMS db migration script to use @fbcnms/sequelize-models 0.1.5 (#5924)

Signed-off-by: GitHub <noreply@github.com>
Signed-off-by: Andrei Lee <andreilee@fb.com>

Co-authored-by: Marie <37634144+themarwhal@users.noreply.github.com>

wallyrb committed:
Add troubleshooting documents to github docs[modified v1.3] (#5078)

Signed-off-by: Wally Rodriguez B <wallyrb@fb.com>

ulaskozat committed:
Bug fix in parsing cause type (#5921)

Signed-off-by: Ulas Kozat <kozat@fb.com>

ymasmoudi committed:
[orc8r][NetworkProbe] Support NetworkProbe APIs in swagger (#4962)

* [orc8r][NetworkProbe] Support NetworkProbe APIs in swagger

Signed-off-by: YOUSSEF EL MASMOUDI <ymasmoudi@fb.com>

* [orc8r][NetworkProbe] add unit tests

Signed-off-by: YOUSSEF EL MASMOUDI <ymasmoudi@fb.com>

* [orc8r][NetworkProbe] move swagger definitions to a separate service

Signed-off-by: YOUSSEF EL MASMOUDI <ymasmoudi@fb.com>

* [orc8r][NetworkProbe] update correlation field

Signed-off-by: YOUSSEF EL MASMOUDI <ymasmoudi@fb.com>

* [orc8r][NetworkProbe] fix nits and change correlation_id type to uint64

Signed-off-by: YOUSSEF EL MASMOUDI <ymasmoudi@fb.com>

* Fix review comments

Signed-off-by: YOUSSEF EL MASMOUDI <ymasmoudi@fb.com>

ulaskozat committed:
Fix bug on referenced value assignment (#5927)

Signed-off-by: Ulas Kozat <kozat@fb.com>

hcgatewood committed:
[codeowners] Bare-metal proposal (#5857)

Signed-off-by: Hunter Gatewood <hgatewood@gmail.com>

tmdzk committed:
Revert changes to push to new registry (#5943)

Signed-off-by: Timothée Dzik <timdzik@fb.com>
